### DISCUSSION TITLE
question: How to customize number of retries after 503 http code

### DISCUSSION THREAD

**Discussion by sundaram-dubey** [apache/superset#24645] [Q&A / Help] [answered]
## Screenshot

My request is retrying 3 time after being timeout once. I want to reduce number of retries to one. How can we do that? 
<img width="1155" alt="Screenshot 2023-07-10 at 5 05 08 PM" src="https://github.com/apache/superset/assets/96574271/949ee5e0-a227-4ad7-854e-4fed15d32e8e">

## Description
I have tried following configuration in config.py, but it didn't work - 

```json
const DEFAULT_FETCH_RETRY_OPTIONS = {
  retries: 1,
  retryDelay: 1000,
  retryOn: [503]
};
```
I don't want to retry sending a request to backend if it fails first time (or first retry) to get the data. It is causing unnecessary load on our databricks warehouse. 

please help here.

**Comment by rusackas** [comment]
That seems like it _ought to_ work, however that config looks like it's a default, which can be overridden by any given call to the SupersetClientClass. Maybe @john-bodley or @nytai know a better way to approach this? BTW, I moved this to a discussion since it's not a bug report, per se.

**Reply by sundaram-dubey** [reply]
Thanks @rusackas, for acknowledgement

**Comment by sundaram-dubey** [comment, marked as answer]
The issue is resolved by overriding SupersetClientClass. 
Slack thread: https://apache-superset.slack.com/archives/CK1P3L23E/p1688989433767779
Thanks.
